cmake/maintainer.cmake

```diff
@@ -31,7 +31,7 @@ SET(MY_CXX_WARNING_FLAGS
 # Extra warning flags for Clang++
 IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   SET(MY_CXX_WARNING_FLAGS
-      "${MY_CXX_WARNING_FLAGS} -Wno-null-conversion -Wno-unused-private-field")
+      "${MY_CXX_WARNING_FLAGS} -Wno-null-conversion -Wno-unused-private-field -Wno-tautological-constant-out-of-range-compare")
```

Comment:
On a second though, please add a comment here so this is removed in an eventual merge to 8.0. This is not required if 5.6 or 5.7 already don't need this

Reply:
This is only required for some assertions: the assertion checks that the actual value (which has an enum type) returned by a function is less than 256. Clang sees that the largest value of the enum is smaller than 256 (as it should be), and reports a warning because of this.

I didn't see this warning in later versions, so  I think these asserts were removed or changed.

Reply:
I misunderstood the PR description: I thought it is the compile_time_assert itself which caused warnings here.

In this case it is those higher version fixes we should take, unless they are intertwined with other major works. It is also an option to remove the tautological asserts.

Reply:
That's a similar, but different warnings, and that was something I could fix with compile-time assertions. 

I wanted to avoid the removal of these asserts, as they are only tautological until they do work - the moment they break, we wouldn't receive this warning.

Reply:
I updated the PR with a version where I removed the two related asserts (unireg.cc and ha_innodb.cc)

Reply:
Would a cast, inside the assert, to a wider type before comparing work?